lib/autokey/gtkui/configwindow.py

```diff
@@ -1360,13 +1371,36 @@ def on_rename(self, widget, data=None):
         dlg.destroy()
 
     def on_treeWidget_row_activated(self, widget, path, viewColumn, data=None):
+        """This function is called when a row is double clicked"""
         if widget.row_expanded(path):
             widget.collapse_row(path)
         else:
             widget.expand_row(path, False)
 
     def on_treeWidget_row_collapsed(self, widget, tIter, path, data=None):
         widget.columns_autosize()
+        p = path.to_string()
+        if len(p) == 1: #closing one of the base dirs
+            self.__hide_row_with_path_up_to(p, 1)
+            return
+        self.__hide_row_with_path_up_to(p, len(p))
+
+    def __hide_row_with_path_up_to(self, pathStr, up_to):
```

Comment:
@sebastiansam55 does this seem like a reasonable function name? I wasn't totally clear on what this code actually as meant to do (differences between trees, rows, and paths etc).

Reply:
a bit verbose but does the job haha

Reply:
Better explicit than implicit. Imo so long as we aren't at Java or Apple C levels, more verbosity is good